online.codingblocks.com/components/AboutCourse/IntroductionCard.vue

```diff
@@ -29,115 +31,191 @@
-                    <img src="https://cb-thumbnails.s3.ap-south-1.amazonaws.com/best-mentors.svg">
+                    <img
+                      src="https://cb-thumbnails.s3.ap-south-1.amazonaws.com/best-mentors.svg"
+                    />
                     <div class="flex-1 white pl-3">
                       <div class="card-sm bold">Best in class Mentors</div>
                     </div>
                   </div>
                 </div>
               </div>
+
               <div class="col-sm-4 col-8">
-                <div class="br-10 border b-white p-4" style="background: rgba(255, 255, 255, 0.1);">
+                <div class="br-10 border b-white p-4" style="background: rgba(255, 255, 255, 0.1)">
                   <div class="row no-gutters align-items-center">
-                    <img src="https://cb-thumbnails.s3.ap-south-1.amazonaws.com/get-hired.svg">
+                    <img
+                      src="https://cb-thumbnails.s3.ap-south-1.amazonaws.com/get-hired.svg"
+                    />
                     <div class="flex-1 white pl-3">
-                      <div class="card-sm bold">Priority Placement in Nagarro</div>
+                      <div class="card-sm bold">
+                        Priority Placement in Nagarro
+                      </div>
                     </div>
                   </div>
                 </div>
               </div>
             </div>
           </div>
-          
-          <!-- <div class="col-6 col-md-3">
-            <div class="info-item">
-            <div>Course Duration </div>
-            <div>6 Months</div>
-          </div> -->
-          </div>
-          
         </div>
       </div>
     </div>
   </div>
 </template>
 
 <script>
+import { mapState } from 'vuex'
+import {JsonApiDataStoreModel} from 'jsonapi-datastore'
 import RatingStars from './RatingStars.vue'
 export default {
+  modules: ['@nuxtjs/axios'],
   name: 'IntroductionCard',
   props: {
     course: {
       type: Object,
-      required: true
+      required: true,
+    },
+  },
+  data() {
+    return {
+      filledHeartClass: 'fas',
+      unfilledHeartClass: 'far',
+      userCourseWishlistModel : null
     }
   },
   components: {
-    RatingStars
+    RatingStars,
   },
   computed: {
     visibleInstructors() {
       return this.course.instructors.slice(0, 2)
     },
-    visibleInstructorNames () {
-      return this.visibleInstructors.map(i => i.name).join(', ')
+    visibleInstructorNames() {
+      return this.visibleInstructors.map((i) => i.name).join(', ')
+    },
+    isCourseWishlisted(){
+      if(this.userCourseWishlistModel!=null){
+        return true;
+      }else{
+        return false;
+      }
     },
-  }
+    ...mapState(['session']),
+  },
+
+  tasks(t){
+      return t(function * isWishListed() {
+        const res =  yield this.$axios.$get(`/courses/${this.course['id']}/relationships/user_course_wishlist`)
+        if(res.data != null){
```

Comment:
we are not supposed to use res.data. use $jsonApiStore.sync(res). this will automatically handle things.